prototype/src/js/DeviceOrienter.jsx

```diff
@@ -0,0 +1,192 @@
+import React from 'react';
+import AlphaOrienter from './AlphaOrienter';
+import BetaOrienter from './BetaOrienter';
+import { CSSTransition } from 'react-transition-group';
+import {
+    delay,
+    calculateNormalizedAlphaOffset,
+    calculateNormalizedBetaOffset
+} from './utils';
+
+export default class DeviceOrienter extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            supportedEventName: null,
+            betaOrientationComplete: false,
+            betaTransionedOut: false,
+            alphaOrientationComplete: false,
+            alphaTransitionedOut: false,
+            firstDeviceOrientationEventReceived: false,
+            alpha: NaN,
+            beta: NaN
```

Comment:
I hacked together a tiny prototype that just logs the values out and routes them through Redux state rather than React state. Both options seem to work pretty well; the Redux version looks something like:

```js
const deviceOrientation = 'deviceorientation';

function startFindOrientation() {
    return {
        type: START_FIND_ORIENTATION,
    };
}

function completeFindOrientation({
    absolute,
    alpha,
    beta,
    gamma,
}) {
    return (dispatch) => {
        dispatch({
            type: COMPLETE_FIND_ORIENTATION,
            payload: {
                absolute,
                alpha,
                beta,
                gamma,
            },
        });
    };
}

export function stopFindOrientation() {
    return (dispatch) => {
        window
            .removeEventListener(
                deviceOrientation,
                e => dispatch(completeFindOrientation(e)),
            );
        dispatch({
            type: FAIL_FIND_ORIENTATION,
        });
    };
}

export function findOrientation() {
    return (dispatch) => {
        dispatch(startFindOrientation());
        window
            .addEventListener(
                deviceOrientation,
                e => dispatch(completeFindOrientation(e)),
            );
    };
}
```